code/modules/mining/mint.dm

```diff
@@ -1,116 +1,193 @@
-/**********************Mint**************************/
-
+#define COIN_COST MINERAL_MATERIAL_AMOUNT * 0.2
 
 /obj/machinery/mineral/mint
 	name = "coin press"
 	icon = 'icons/obj/economy.dmi'
-	icon_state = "coinpress0"
+	icon_state = "coin_press"
 	density = TRUE
 	anchored = TRUE
-	var/newCoins = 0   //how many coins the machine made in it's last load
-	var/processing = FALSE
-	var/chosen = MAT_METAL //which material will be used to make coins
-	var/coinsToProduce = 10
-	speed_process = TRUE
-
+	/// How many coins did the machine make in total.
+	var/total_coins = 0
+	/// Is it creating coins now?
+	var/active = FALSE
+	/// Which material will be used to make coins or for ejecting.
+	var/chosen_material
+	/// Inserted money bag.
+	var/obj/item/storage/bag/money/money_bag
 
 /obj/machinery/mineral/mint/Initialize(mapload)
 	. = ..()
-	AddComponent(/datum/component/material_container, list(MAT_METAL, MAT_PLASMA, MAT_SILVER, MAT_GOLD, MAT_URANIUM, MAT_DIAMOND, MAT_BANANIUM, MAT_TRANQUILLITE), MINERAL_MATERIAL_AMOUNT * 50, FALSE, /obj/item/stack)
+	var/static/list/coin_materials = list()
+	if(!length(coin_materials))
+		for(var/datum/material/coin_mat as anything in subtypesof(/datum/material))
+			var/obj/item/coin/coin_type = coin_mat.coin_type
+			if(!coin_type)
+				continue
+			coin_materials += coin_mat.id
+
+	AddComponent(/datum/component/material_container, coin_materials, MINERAL_MATERIAL_AMOUNT * 50, FALSE, /obj/item/stack, _after_insert=CALLBACK(src, PROC_REF(material_insert)))
+	chosen_material = pick(coin_materials[1])
+
+/obj/machinery/autolathe/Destroy()
```

Comment:
```suggestion
/obj/machinery/mineral/mint/Destroy()
```
Looks like you forgot to change this when copy and pasting.

Reply:
Looks like yes